vividus-plugin-azure-service-bus/src/main/java/org/vividus/azure/servicebus/service/ServiceBusService.java

```diff
@@ -17,57 +17,153 @@
         }
         catch (ServiceBusException e)
         {
             throw new ServiceBusSendMessageException("Unable to send message to Azure Service Bus", e);
         }
     }
+
+    public void startConsuming(String clientKey)
+    {
+        BlockingQueue<ServiceBusReceivedMessage> messages = new LinkedBlockingDeque<>();
+        testContext.get(MESSAGES_KEY, HashMap::new).put(clientKey, messages);
+        ServiceBusClientEntity serviceBusClientEntity = getServiceBusClientConfig(clientKey);
+
+        String namespace = serviceBusClientEntity.getNamespace();
+        ServiceBusClientBuilder.ServiceBusProcessorClientBuilder processorBuilder = clientBuilder
+                .fullyQualifiedNamespace(namespace + DEFAULT_NAMESPACE_POSTFIX)
+                .processor()
+                .processMessage(context -> {
+                    ServiceBusReceivedMessage message = context.getMessage();
+                    BinaryData messageBody = message.getBody();
+                    LOGGER.info("Processing message. Session: {}, Sequence #: {}. Contents: {}", message.getMessageId(),
+                            message.getSequenceNumber(), messageBody);
+                    messages.add(message);
+                })
+                .processError(context ->
+                {
+                    throw (ServiceBusException) context.getException();
+                });
+        ChannelType channelType = serviceBusClientEntity.getChannelType();
+        String name = serviceBusClientEntity.getName();
+        if (channelType == ChannelType.QUEUE)
+        {
+            processorBuilder.queueName(name);
+        }
+        else
+        {
+            processorBuilder.topicName(name);
+            processorBuilder.subscriptionName(serviceBusClientEntity.getSubscriptionName());
+        }
+        ServiceBusProcessorClient client = processorBuilder.buildProcessorClient();
+
+        clients.put(clientKey, client);
+        LOGGER.info("'{}' Azure Service Bus Consumer of messages from the {} '{}' in namespace {} is started",
+                clientKey, channelType, name, namespace);
+        client.start();
+    }
+
+    private ServiceBusClientEntity getServiceBusClientConfig(String clientKey)
+    {
+        return clientConfigs.get(clientKey, "No Service Bus client with key `%s` is configured", clientKey);
+    }
+
+    public void stopConsuming(String clientKey)
+    {
+        ServiceBusProcessorClient client = clients.remove(clientKey);
+        if (client != null)
+        {
+            LOGGER.info("'{}' Azure Service Bus Consumer is stopped", clientKey);
+            client.close();
+        }
+        else
+        {
+            throw new IllegalArgumentException(
+                    "There are no running Azure Service Bus consumers associated with the key:" + clientKey);
+        }
+    }
+
+    public void stopAll()
+    {
+        clients.keySet().stream().toList().forEach(this::stopConsuming);
```

Comment:
forEach can be invoked on stream

Reply:
I use remove method over the clients map. Usage of forEach will lead to ConcurrentModificationException